pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -171,6 +174,10 @@ func New(
 	ctrl.clusterVersionListerSynced = clusterVersionInformer.Informer().HasSynced
 
 	ctrl.patchContainerRuntimeConfigsFunc = ctrl.patchContainerRuntimeConfigs
+	// Add to the queue to trigger a sync when an upgrade happens
+	// this ensures that the capabilities MC is created on an upgrade
+	// This will be removed in the next version
+	ctrl.queue.Add("force-sync-on-upgrade")
```

Comment:
@yuqi-zhang, this line adds an item to the controller queue to force it to resync on an upgrade/start up. For a fresh installation, this was causing failures as the bootstrap didn't have the capabilities MC, but was expecting it I guess and was complaining about an expected rendered MC not being found. So I added the `RunCapBootstrap` function at line https://github.com/openshift/machine-config-operator/pull/2050/files#diff-22412a83973655ff84984dc7d9969013R699. 
I tested this out and had a successful installation, but want to double check if there is a case where having this force sync on controller start up could cause issues with the bootstrap logic. wdyt?

Reply:
Hmm sorry I missed this sync when I did the initial review. Is this sync needed to get the capabilities MC to trigger?

When an upgrade happens the new MCO with this new CRCC should be deployed first. It will then (when running the sync) call `createCapabilitiesMC` and split out the new MC before it attempts an update. The bootstrap should also be running a MCC which should also have the same workflow. Thus as soon as you install/update to a version with this patch, it would create the MC + configmap without the need for a sync again right? Maybe I am confused there

Reply:
So when you don't have any ctrcfg CRs created in your cluster and you upgrade, the controller does not resync and that is why I added an item to the queue to force this sync to happen. So when the cluster is upgraded and the controller starts up, it has an item in its queue and we sync. This sync splits creates the capabilities MC + config map.
For the bootstrap, we create the capabilities MC but not the config map, because it looks like when the cluster is ready and the controller starts up, a sync is forced by this line. In my test run everything worked fine, the capabilities MCs and config map were created.
I am not very familiar with the bootstrap process, so wanted to see if there may be any scenarios where having the controller sync up immediately after the bootstrap/installation could cause any issues.

Reply:
Ah I see, hmm, I almost feel like either
a) the crcc is updated to syncs this
b) this checking lives somewhere else instead (maybe the MCC directly? Not sure if that'd be a good idea either)

Offhand I don't know if this will cause bootstrap problems but since the installation is successful now I think it shouldnt? This sync really just affects the CRCC anyways right

Reply:
Yeah, from my tests the sync right after installation hasn't affected anything with the bootstrap stuff, so we should be good.